packages/caliper-core/lib/manager/orchestrators/round-orchestrator.js

```diff
@@ -225,14 +231,24 @@ class RoundOrchestrator {
             }
         }
 
-        // clean up, with "silent" failure handling
         try {
             this.report.printResultsByRound();
             await this.report.finalize();
         } catch (err) {
             logger.error(`Error while finalizing the report: ${err.stack || err}`);
         }
 
+        await this._cleanup();
+
+        let benchEndTime = Date.now();
+        logger.info(`Benchmark finished in ${(benchEndTime - benchStartTime)/1000.0} seconds. Total rounds: ${success + failed}. Successful rounds: ${success}. Failed rounds: ${failed}.`);
+    }
+
+    /**
+     * Stops the benchmark.
+     */
+    async _cleanup() {
```

Comment:
We should call this `stop` for caliper-engine to call, but it must also call `this.workerOrchestrator.stop` otherwise it won't send the exit message to workers which is what should terminate the workers. 
eg
```suggestion
    async stop() {
        try {
            await this.monitorOrchestrator.stopAllMonitors();
        } catch (err) {
            logger.error(`Error while stopping monitors: ${err.stack || err}`);
        }

        try {
            await this.workerOrchestrator.stop();
        } catch (err) {
            logger.error(`Error while stopping workers: ${err.stack || err}`);
        }
    }
```